shared/common-adapters/tab-bar.native.js

```diff
@@ -51,19 +48,10 @@ const TabBarButton = (props: TabBarButtonProps) => {
   const badgeNumber = props.badgeNumber || 0
 
   let badgeComponent = null
-  if (badgeNumber > 0) {
+  if (props.badgeNumber) {
```

Comment:
The way the tab bar component is set up also seems like it could be refactored a little bit – I didn't want to go too in depth here but definitely something that could be hit in a future ticket.